client/components/domains/register-domain-step/index.jsx

```diff
@@ -337,9 +337,26 @@ class RegisterDomainStep extends Component {
 		return strippedHostname ?? hostname;
 	}
 
+	getInitialQueryInAIFlow() {
+		if ( this.props.flowName !== 'ai-site-builder' ) {
+			return;
+		}
+
+		// TODO: Maybe reading from meta or url would be useful too.
+		if ( ! this.props.selectedSite || ! this.props.selectedSite.name ) {
+			return;
+		}
+
+		return this.props.selectedSite.name;
```

Comment:
```suggestion
		// TODO: Maybe reading from meta or url would be useful too.
		return this.props.selectedSite?.name;
```

Reply:
[Jynx](https://github.com/Automattic/wp-calypso/pull/100948#discussion_r1995296660) !

In case returning `undefined` instead of `""` was important, we should also `|| undefined` at the end of the statement

Reply:
https://github.com/Automattic/wp-calypso/pull/100948/commits/82ef7eb326213c02f38296ab314a690ba9e8b7cf